### PR TITLE
ALP-63: Implement volunteer sign up form validation

### PR DESCRIPTION
**Pull request by thaingocnguyen**

### CODE REVIEW

File: aunt-leahs-app/src/components/CustomButton.jsx [hunk 1]
```diff
@@ -38,19 +38,26 @@ const useStyles = makeStyles({
 
 // Two different sizes: small and large 
 // Two different colors: primary (teal) and secondary (red)
-const CustomButton = ({ size, color, onClick, children }) => {
+const CustomButton = ({ size, color, onClick, children, type, disabled }) => {
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/CustomButton.jsx]: I generally name my booleans with an `is` or `has` prefix. In this case, I would suggest `isDisabled` here?

File: aunt-leahs-app/src/components/TextInput.jsx [hunk 2]
```diff
@@ -13,10 +13,13 @@ const useStyles = makeStyles({
...
         padding: '0px 5px',
     },
+    inputError: {
+        border: '2px solid red'
+    }
 });
 
 
-const TextInput = ({ title, size, onChange, value, type }) => {
+const TextInput = ({ id, title, size, onChange, value, type, error, required, onBlur, helperText }) => {
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/TextInput.jsx]: Similar to the other comment, I would suggest `isRequired` here

File: aunt-leahs-app/src/components/TextInput.jsx [hunk 3]
```diff
@@ -26,23 +29,35 @@ const TextInput = ({ title, size, onChange, value, type }) => {
...
-                <TextField classes={{ root: classes.root }}
-                    InputProps={{ disableUnderline: true }}
-                    placeholder={' ' + title}
-                    margin='none'
-                    fullWidth='True'
-                    onChange={onChange}
-                    type={type}
-                    value={value} />
-            </Typography>
+        <div className={`${length} textInput`}>
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/TextInput.jsx]: I'm trying to standardize all the CSS in ALP-60, try to use kebab case for css class names (i.e. `text-input` here). I'm pretty sure that's the convention from what I see on mdn

File: aunt-leahs-app/src/constants.jsx [hunk 4]
```diff
@@ -103,3 +103,5 @@ export const pages = {
 	VOLUNTEER_CHECK_IN: 'Volunteer Check In',
 	VOLUNTEER_SIGN_UP: 'Volunteer Sign Up'
 };
+
+export const requiredText = "Required";
```
Comment (justinkkwan) [inline aunt-leahs-app/src/constants.jsx]: Add new line

File: aunt-leahs-app/src/components/VolunteerSignUpForm.jsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+    handleSubmit: (values, { setSubmitting }) => {
+        setTimeout(() => {
+            alert(JSON.stringify(values, null, 2));
+            setSubmitting(false);
+        }, 1000)
+    }
+})(VolunteerSignUpFormContent);
+
+export default VolunteerSignUpForm;
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/VolunteerSignUpForm.jsx]: Needs new line

File: aunt-leahs-app/src/components/TextInput.jsx (hunk 2 above)
Comment (justinkkwan) [reply aunt-leahs-app/src/components/TextInput.jsx]: And maybe `hasError`?

**Review by justinkkwan — APPROVED** [review]
I left some naming suggestions for clarity and missing new line comments. Also, before you merge, please let me know because in my PR I remove the styles.css file and put everything into App.css (also I got rid of all the makeStyles)

File: aunt-leahs-app/src/components/CustomButton.jsx (hunk 1 above)
Comment (thaingocnguyen) [reply aunt-leahs-app/src/components/CustomButton.jsx]: Usually I do that as well, but in this case I was just matching it to the props of the material ui button for consistency. I'll change it to isDisabled.

**Comment by thaingocnguyen** [comment]
I've addressed Justin's code review comments

File: aunt-leahs-app/src/styles.css [hunk 6]
```diff
@@ -514,4 +531,4 @@ a {
   color: #00A19B;
   cursor: pointer;
   border-bottom: solid 2px #00A19B;
-}
+}
```
Comment (Leonlui) [inline aunt-leahs-app/src/styles.css]: newline

**Review by Leonlui — APPROVED** [review]
🥇

**Pull request merged**